http/handlers.go

```diff
@@ -96,13 +96,22 @@ func RegisterHandlers(mux httpmux.Mux, containerManager manager.Manager, httpAut
 // the provided HTTP mux to handle the given Prometheus endpoint.
 func RegisterPrometheusHandler(mux httpmux.Mux, containerManager manager.Manager, prometheusEndpoint string,
 	f metrics.ContainerLabelsFunc, includedMetrics container.MetricSet) {
-	r := prometheus.NewRegistry()
-	r.MustRegister(
-		metrics.NewPrometheusCollector(containerManager, f, includedMetrics),
-		prometheus.NewGoCollector(),
-		prometheus.NewProcessCollector(os.Getpid(), ""),
-	)
-	mux.Handle(prometheusEndpoint, promhttp.HandlerFor(r, promhttp.HandlerOpts{ErrorHandling: promhttp.ContinueOnError}))
+	mux.Handle(prometheusEndpoint, http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
```

Comment:
Does this recreate the PrometheusCollector each time a request comes in?  Can we avoid this?

Reply:
Sorry for the my own delay :cry: !
Yes it does. In general i think its not costly, but i think we can also just initialize the collector outside of the handler, register with the registry, and then customize the option ( perhaps with new exported option like `func (c *PrometheusCollector) SetRequestOptions(opt)` before running `promhttp.HandlerFor` which triggers `Collect` for each registered collector.

Reply:
I would prefer a solution that does something like that to the current solution